liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/PluginConfigSupport.java

```diff
@@ -33,28 +34,28 @@
 public class PluginConfigSupport extends StartDebugMojoSupport {
     
     /**
-     * Packages to install. One of "all", "dependencies" or "project".
+     * Application directory. 
      */
-    @Parameter(property = "installAppPackages", defaultValue = "dependencies", readonly = true)
-    private String installAppPackages = "dependencies";
+    @Parameter(property = "appsDirectory", defaultValue = "dropins")
+    protected String appsDirectory;
     
     /**
-     * Application directory.
+     * Strip version. 
      */
-    @Parameter(property = "appsDirectory", defaultValue = "dropins", readonly = true)
-    private String appsDirectory = "dropins";
+    @Parameter(property = "stripVersion", defaultValue = "false")
+    protected boolean stripVersion;
     
     /**
-     * Strip version.
+     * Loose application. 
      */
-    @Parameter(property = "stripVersion", defaultValue = "false", readonly = true)
-    private boolean stripVersion = false;
+    @Parameter(property = "looseApplication", defaultValue = "false")
+    protected boolean looseApplication;
     
     /**
-     * Loose application. 
+     * Packages to install. One of "all", "dependencies" or "project".
      */
-    @Parameter(property = "looseApplication", defaultValue = "false", readonly = true)
-    private boolean looseApplication;
+    @Parameter(property = "installAppPackages", defaultValue = "dependencies")
+    protected String installAppPackages;
```

Comment:
I was using this annotation incorrectly before. The explanation of @readonly can be found in https://maven.apache.org/developers/mojo-api-specification.html#The_Descriptor_and_Annotations